lab-bundle-spec-v1.md

```diff
@@ -0,0 +1,248 @@
+
+```yml
+environment_resources:
+  - type: gcp_project
+    id: my_primary_project
+    dm_template:
+      script: deployment_manager
+  - type: gcp_user
+    id: primary_user
+    permissions:
+      - project: my_primary_project
+        roles:
+          - roles/editor
+          - roles/bigquery.admin
+  - type: gcp_user
+    id: secondary_user
+```
+
+#### Valid types
+
+##### GCP Project (gcp_project)
+
+attribute                     | required | type    | notes
+----------------------------- | -------- | ------- | --------------------------------------
+dm_template.script            |          | path    | Relative path to a Deployment Manager directory tree.
+dm_template.custom_properties |          | array   | Array of key/value pairs.
+fleet                         |          | enum*   | Specify a Qwiklabs fleet to pull the project from.
+
+```yml
+- type: gcp_project
+  id: secondary_project
+  fleet: gcpfree
+  dm_template:
+    script: deployment_manager
+    custom_properties:
+      - key: userNameWindows
+        value: student
+```
+
+> **NOTE:** Not all GCP fleet names are supported.
+>
+> The existing concept of Qwiklabs' Fleets does not have a single analog in
+> content bundles. Notice that some fleet types map to resource types (e.g.
+> `gsuite_multi_tenant` fleet is now the `gsuite-domain` resource type).
+> However other fleets are "variants" of the same resource type (e.g.
+> "free", "ASL", and "standard" GCP projects). Therefore, not all fleet names that
+> an experienced Qwiklabs author may be familiar with, are allowed in the `fleet`
+> field.
+>
+> Presently, authors are allowed to specify one of the following fleet types for
+> a gcp_project:
+> - gcpd [default]
+> - gcpfree
+> - gcpasl
+>
+> Future versions of the Content Bundle spec may use different terminology for
+> resource_type variations to avoid conflation.
+
+##### GCP User (gcp_user)
+
+attribute   | required | type       | notes
+----------- | -------- | ---------- | ----------------------------------------
+permissions |          | array      | Array of project/roles(array) pairs
+
+```yml
+  - type: gcp_user
+    id: primary_user
+    permissions:
+      - project: my_primary_project
+        roles:
+          - roles/editor
+          - roles/bigquery.admin
+```
+
+##### GSuite Domain (gsuite_domain)
+
+attribute   | required | type       | notes
+----------- | -------- | ---------- | ----------------------------------------
+No additional attributes
+
+```yml
+- type: gsuite_domain
+  id: primary_domain
+```
+
+##### Future Resource Types
+
+- AWS Account (aws-account)
+- iPython Notebook (ipython-notebook)
+
+> **NOTE:** A draft of the `aws-account` resource type was previously specified
+> in this document. See [previous version](https://github.com/CloudVLab/qwiklabs-content-bundle-spec/blob/93896ced4ae5b543132d7a10d838ac17bd5ae3e1/lab-bundle-spec.md) for details.
+
+### Activity Tracking (Alpha)
```

Comment:
Do we want to better codify this in V2 or is that a huge task that is beyond the scope of this PR?

Reply:
Unless someone is well on their way to figuring out what this would look like for V2, I think we need to punt it to the next version.

Reply:
Unfortunately, the latter.